public/app/plugins/datasource/tempo/NativeSearch.tsx

```diff
@@ -43,22 +44,51 @@ const NativeSearch = ({ datasource, query, onChange, onBlur, onRunQuery }: Props
   const [hasSyntaxLoaded, setHasSyntaxLoaded] = useState(false);
   const [autocomplete, setAutocomplete] = useState<{
     serviceNameOptions: Array<SelectableValue<string>>;
+    selectedServiceName: SelectableValue<string> | undefined;
     spanNameOptions: Array<SelectableValue<string>>;
+    selectedSpanName: SelectableValue<string> | undefined;
   }>({
     serviceNameOptions: [],
+    selectedServiceName: undefined,
     spanNameOptions: [],
+    selectedSpanName: undefined,
   });
 
+  const fetchServiceNameOptions = useMemo(
```

Comment:
we could use `useDebounce` from react-use instead, it would make the code a bit easier to read.

Reply:
Unless I'm misunderstanding the docs, it looks like you `useDebounce` works like useEffect. It doesn't return a function that can be called. Instead, it runs whenever one of its dependencies changes. For that to work here, I'd need to add additional state, update that when the menu opens, then that state change would trigger the debounce.

Reply:
hmm you are right, seems I read it wrong. Yeah then ignore the suggestion. In that case though would it make sense to move it into a separate hook? Seems like the only difference between those 2 is `'service.name'` string and `serviceNameOptions` key to assign the results.

Reply:
Yeah, it is repetitive. I'll address that in one of my next PRs for search (I have a catch-all one I'm leaving for last)